protocol/common/message.go

```diff
@@ -325,11 +330,34 @@ func extractMentions(parsedText ast.Node) []string {
 	return visitor.mentions
 }
 
+type LinkNodeVisitor struct {
```

Comment:
You probably want to modify the existing class, above, to also save links, in the same way you did, so we avoid vising each node twice

Reply:
Good idea, thanks! 
It is still wip, I'll re-request review when ready :)